### PR TITLE
Add manual container instantiation sample

### PR DESCRIPTION
**Pull request by mcruzdev**
## :loudspeaker: Type of change

- [ ] Bugfix
- [ ] New feature
- [x] Enhancement
- [ ] Refactoring

## :scroll: Description

This pull request adds a new sample for SQS, in other words this add a sample showing how can use manual container instantiation with SQS.

initially this pull request was on this [pull request](https://github.com/awspring/spring-cloud-aws/pull/774).

It is a new pull request for issue #670 
## :bulb: Motivation and Context

The motivation is to make a better documentation with code samples.

## :pencil: Checklist

- [ ] I reviewed submitted code
- [ ] I added tests to verify changes
- [ ] I updated reference documentation to reflect the change
- [ ] All tests passing
- [ ] No breaking changes

### CODE REVIEW

File: spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+		return args -> sqsTemplate.send(to -> to.queue(NEW_USER_QUEUE)
+			.payload(new User(UUID.randomUUID(), "John"))
+		);
+	}
+
+	@Bean
+	public SqsTemplate sqsTemplateManualContainerInstantiation(SqsAsyncClient sqsAsyncClient) {
+		return SqsTemplate.builder()
+			.sqsAsyncClient(sqsAsyncClient)
+			.configure(options -> options.acknowledgementMode(TemplateAcknowledgementMode.MANUAL))
```
Comment (tomazfernandes) [inline spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java:55]: Same as the other one - doesn't make a difference if we're not receiving messages with the Template.

File: spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+			.build();
+	}
+
+	@Bean
+	SqsMessageListenerContainer<User> sqsMessageListenerContainer(SqsAsyncClient sqsAsyncClient) {
+		SqsMessageListenerContainer<User> container = new SqsMessageListenerContainer<>(sqsAsyncClient);
+		container.setMessageListener((message) -> {
+			LOGGER.info("Received message {}", message);
+			Acknowledgement.acknowledge(message);
```
Comment (tomazfernandes) [inline spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java:64]: Given we already have a sample for `MANUAL` ack, how about we leave this one with default ack settings?

File: spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		});
+		container.setQueueNames(NEW_USER_QUEUE);
+		container.setAcknowledgementResultCallback(new AckResultCallback());
+		container.configure(sqsContainerOptionsBuilder ->
+			sqsContainerOptionsBuilder
+				.acknowledgementMode(AcknowledgementMode.MANUAL));
+		return container;
+	}
+
+	static class AckResultCallback implements AcknowledgementResultCallback<User> {
```
Comment (tomazfernandes) [inline spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java:74]: I like the idea of having the `AckResultCallback`, but how about we use it in the manual ack sample?

This way we focus on the Container instantiation for this one, and leave ack related stuff for the other one.

Makes sense?

File: spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+		container.setMessageListener((message) -> {
+			LOGGER.info("Received message {}", message);
+			Acknowledgement.acknowledge(message);
+		});
+		container.setQueueNames(NEW_USER_QUEUE);
+		container.setAcknowledgementResultCallback(new AckResultCallback());
+		container.configure(sqsContainerOptionsBuilder ->
+			sqsContainerOptionsBuilder
+				.acknowledgementMode(AcknowledgementMode.MANUAL));
+		return container;
```
Comment (tomazfernandes) [inline spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java:71]: We have an awesome `builder` for the container - how about we show it off? 😄

**Review by tomazfernandes — CHANGES_REQUESTED** [review]
Same, thanks again @mcruzdev! Let me know if the comments make sense to you.

File: spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java (hunk 4 above)
Comment (mcruzdev) [reply spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java:71]: hello @tomazfernandes, sorry for the delay.

With builder I need to return a `SqsMessageListenerContainer<Object>` of `Object` not of `User`.

```java
    @Bean
    SqsMessageListenerContainer<Object> sqsMessageListenerContainer(SqsAsyncClient sqsAsyncClient) {
        return SqsMessageListenerContainer.builder()
                .sqsAsyncClient(sqsAsyncClient)
                .queueNames(NEW_USER_QUEUE)
                .messageListener((message) -> {
                    LOGGER.info("Received message {}", message);
                })
                .build();
    }
```

Is there another way to do it?

Comment (mcruzdev) [reply spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SqsManualContainerInstantiationSample.java]: Hi @tomazfernandes, how are you doing?

Sorry for the delay!

I made some changes, thank you for the review

**Comment by mcruzdev** [comment]
Hi @tomazfernandes , can you review this pull request?

**Review by tomazfernandes — APPROVED** [review]
Looks great @mcruzdev, thanks!

**Pull request merged**